MarvelApiClientTests/CharactersApiClientTests.swift

```diff
@@ -0,0 +1,105 @@
+            .andReturn(200)
+            .withBody(fromJsonFile("getCharacterById"))
+
+        let result = charactersApiClient.getById("1")
+
+        expect(result).toEventually(beGetCharacterByIdSuccess())
+        assertContainsExpectedCharacterDTO(result.value!)
+    }
+
+    private func givenACharactersApiClient() -> CharactersApiClient {
+        return CharactersApiClient(baseEndpoint: marvelBaseEndpoint,
+            timeProvider: timeProvider,
+            httpClient: httpClient, parser: CharactersParser())
+    }
+
+    private func assertContainsExpectedGetCharactersDTO(getCharactersDTO: GetCharactersDTO) {
+        expect(getCharactersDTO.count).to(equal(1))
+        expect(getCharactersDTO.offset).to(equal(0))
+        expect(getCharactersDTO.limit).to(equal(1))
+        expect(getCharactersDTO.characters[0].id).to(equal("1011334"))
+        expect(getCharactersDTO.characters[0].name).to(equal("3-D Man"))
+        expect(getCharactersDTO.characters[0].description).to(equal("Tridimensional Hero"))
+        expect(getCharactersDTO.characters[0].thumbnail.path).to(equal("http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784"))
+        expect(getCharactersDTO.characters[0].thumbnail.format).to(equal("jpg"))
+        expect(getCharactersDTO.characters[0].thumbnail.fullPath).to(equal("http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784.jpg"))
+        expect(getCharactersDTO.characters[0].comics.count).to(equal(11))
+        expect(getCharactersDTO.characters[0].comics[0].name).to(equal("Avengers: The Initiative (2007) #14"))
+        expect(getCharactersDTO.characters[0].series.count).to(equal(2))
+        expect(getCharactersDTO.characters[0].series[0].name).to(equal("Avengers: The Initiative (2007 - 2010)"))
+        expect(getCharactersDTO.characters[0].stories.count).to(equal(17))
+        expect(getCharactersDTO.characters[0].stories[0].name).to(equal("Cover #19947"))
+        expect(getCharactersDTO.characters[0].stories[0].type).to(equal("cover"))
+        expect(getCharactersDTO.characters[0].events.count).to(equal(1))
+        expect(getCharactersDTO.characters[0].events[0].name).to(equal("Secret Invasion"))
+    }
+
+    private func beGetCharactersDTOSuccess<T>() -> MatcherFunc<T?> {
+        return MatcherFunc { actualExpression, failureMessage in
+            failureMessage.postfixMessage = "be success"
+            let future = try actualExpression.evaluate() as! Future<GetCharactersDTO,NSError>
+            return future.isSuccess
+        }
+    }
+
+    private func beGetCharacterByIdSuccess<T>() -> MatcherFunc<T?> {
+        return MatcherFunc { actualExpression, failureMessage in
+            failureMessage.postfixMessage = "be success"
+            let future = try actualExpression.evaluate() as! Future<CharacterDTO,NSError>
+            return future.isSuccess
+        }
+    }
```

Comment:
@davideme can you help me to refactor these two methods to be the same waiting for a future success?